File: DSA/Vaishnavi_590017838/Day_34/Question1.cpp
class MyQueue {
public:

    stack<int> input;
    stack<int> output;
    MyQueue() {
    }
    void push(int x) {
        input.push(x);
    }
    int pop() {
        if(output.empty()) {
            while(!input.empty()) {
                output.push(input.top());
                input.pop();
            }
        }
        int value = output.top();
        output.pop();
        return value;
    }
    int peek() {
        if(output.empty()) {
            while(!input.empty()) {
                output.push(input.top());
                input.pop();
            }
        }
        return output.top();
    }
    bool empty() {
        return input.empty() && output.empty();
    }
};

File: DSA/ShlokMadan_590017851/Day_34/Question1.java
import java.util.Stack;

public class Question1
{
    Stack<Integer> in;
    Stack<Integer> out;

    public Question1()
    {
        in=new Stack<>();
        out=new Stack<>();
    }

    public void push(int x)
    {
        in.push(x);
    }

    public int pop()
    {
        peek();
        return out.pop();
    }

    public int peek()
    {
        if(out.isEmpty())
        {
            while(!in.isEmpty())
            {
                out.push(in.pop());
            }
        }
        return out.peek();
    }

    public boolean empty()
    {
        return in.isEmpty() && out.isEmpty();
    }

    public static void main(String[] args)
    {
        Question1 q=new Question1();
        q.push(1);
        q.push(2);

        System.out.println(q.peek());
        System.out.println(q.pop());
        System.out.println(q.empty());
    }
}